### PR TITLE
Test Where TypeSupport is Used as a Plugin

### PR DESCRIPTION
**Pull request by iguessthislldo**

### CODE REVIEW

File: tests/DCPS/TypeSupportPlugin/TypeSupportPlugin.cpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  DDS::DomainParticipantFactory_var dpf = TheParticipantFactoryWithArgs(argc, argv);
+
+  ACE_DLL ts_plugin;
+  if (ts_plugin.open(ACE_TEXT("ConsolidatedMessengerIdl"))) {
+    ACE_ERROR((LM_ERROR, "ERROR: Failed to open the type support library\n"));
+    return 1;
+  }
+
+  const char* type_name = "Messenger::Message";
+  printf("Getting TypeSupport for %s\n", type_name);
```
Comment (jwillemsen) [inline tests/DCPS/TypeSupportPlugin/TypeSupportPlugin.cpp:18]: Why not use ACE_DEBUG?

File: tests/DCPS/TypeSupportPlugin/TypeSupportPlugin.cpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  DDS::TypeSupport_ptr ts = Registered_Data_Types->lookup(0, type_name);
+  if (!ts) {
+    ACE_ERROR((LM_ERROR, "ERROR: TypeSupport wasn't registered!\n"));
+    return 1;
+  }
+
+  CORBA::String_var ts_name = ts->get_type_name();
+  printf("Name according to type support is %s\n", ts_name.in());
```
Comment (mitza-oci) [inline tests/DCPS/TypeSupportPlugin/TypeSupportPlugin.cpp:26]: Should the test fail if the name wasn't expected?
Any other verification steps like using the TypeSupport to get the DynamicType?

File: tests/DCPS/TypeSupportPlugin/run_test.pl [hunk 3]
```diff
@@ -0,0 +1,16 @@
+eval '(exit $?0)' && eval 'exec perl -S $0 ${1+"$@"}'
```
Comment (mitza-oci) [inline tests/DCPS/TypeSupportPlugin/run_test.pl:1]: Can this be added to a test .lst file?

**Review by mitza-oci — CHANGES_REQUESTED** [review]

**Review by mitza-oci — APPROVED** [review]

File: tests/DCPS/TypeSupportPlugin/TypeSupportPlugin.mpc [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  Source_Files {
+  }
+}
+
+project(TypeSupportPluginUser): dcps_test {
+  exename = TypeSupportPluginUser
+  Source_Files {
+    TypeSupportPlugin.cpp
+  }
+}
```
Comment (mitza-oci) [inline tests/DCPS/TypeSupportPlugin/TypeSupportPlugin.mpc:16]:
```suggestion
  IDL_Files {
  }
}
```
Defaults to `*.idl` which doesn't seem to be what we want here

**Review by mitza-oci — CHANGES_REQUESTED** [review]

**Review by mitza-oci — APPROVED** [review]
Ready to merge once this CI error is fixed https://github.com/objectcomputing/OpenDDS/actions/runs/3496083930/jobs/5854701992#step:16:17

**Pull request merged**